src/main/java/com/amplitude/api/DatabaseHelper.java

```diff
@@ -377,7 +369,55 @@ protected synchronized List<JSONObject> getEventsFromTable(
             }
             close();
         }
-        return events;
+
+        try {
+            List<JSONObject> events = new LinkedList<JSONObject>();
+            for (Long eventId : eventIds) {
+                JSONObject event = getEventFromTable(table, eventId);
+                if (event != null) {
+                    events.add(event);
+                }
+            }
+            return events;
+        } finally {
+            close();
+        }
+    }
+
+    protected synchronized JSONObject getEventFromTable(String table, long eventId) throws JSONException {
+        JSONObject event = null;
+        Cursor cursor = null;
+        try {
+            SQLiteDatabase db = getReadableDatabase();
```

Comment:
@falconandy do you think the extra DB setup/teardown could have other affects on performance?

Since we have the other areas with this issue handled, I would prefer to keep the previous `getEventsFromTable()` logic for the common case, and use this as a fallback for errors in [getTransferIdentifyEvent](https://github.com/amplitude/Amplitude-Android/blob/main/src/main/java/com/amplitude/api/IdentifyInterceptor.java#L118). We could make a new `dbHelper.getIdentifyInterceptorsLowMemory()` and use it on CursorWindow error.

I agree, I don't think we can delay the identify information. Maybe if intercepted identify events fail to load with low memory version we should drop them from the DB? wdyt?

Reply:
@justin-fiedler 
> do you think the extra DB setup/teardown could have other affects on performance?

Yes, it could add minor slowness. I've modified code to use original `getEvents` implementation and to switch to fallback `row-by-row` load if `CursorWindowAllocationException` raises. I think, it could help for events, identifies and intercepted identifies (instead of deferring).

>  Maybe if intercepted identify events fail to load with low memory version we should drop them from the DB?

I believe the fallback `row-by-row` schema should help in most cases except a single intercepted identify with huge payload. Yes, we could drop "huge" intercepted identifies as a last resort (with additional logging).